File: final/Trabalho_API_web/Models/Modelo.cs
using System.ComponentModel.DataAnnotations;

public class Modelo
{
    [Key]
    public int? idModelo {get; set;}
    public int? idMarca {get;set;}
    public Marca? Marca {get; set;}
    public string? nomeModelo {get; set;}
    public int? AnoModelo {get; set;}
    public enum segmento {automovel, moto, onibus, caminhão}
    public segmento Tipo {get; set;}
    public enum Tamanho {hatch, sedã, suv, picape, perua, minivan, esportivo, furgão}
    public Tamanho Porte {get; set;}
}

File: final/Trabalho_APP_web/src/app/Modelo.ts
import { Marca } from "./Marca";

export class Modelo {
    _idModelo: number = 0;
    _nomeModelo: String = "";
    _AnoModelo: number = 0;
    _segmento: String ="";
    _TipoModelo: String = "";
    _idMarca: Marca | undefined;
}